drivers/scheduler.c

```diff
@@ -115,17 +115,17 @@ scheduler_prepare_events(scheduler_t *s)
 		if (event->masked || event->dead)
 			continue;
 
-		if (event->mode & SCHEDULER_POLL_READ_FD) {
+		if (event->mode & SCHEDULER_POLL_READ_FD && event->fd > 0) {
```

Comment:
Should this be "eventfd >= 0" ?

(likewise below)

Reply:
yes

Reply:
Done